### PR TITLE
[Feature] Add new ECG delineator method: Peak-Prominence (Emrich et al., 2024)

### PR DESCRIPTION
**Pull request by JonasEmrich**
# Description
This PR adds the `prominence` delineator to the `ecg_delineate` method. The prominence delineator tries to achieve a robust and interpretable wave detection by building upon peak prominence and physiology-based search windows. [1]

# Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/neuropsychology/NeuroKit/blob/master/.github/CONTRIBUTING.rst#structure-and-code) file.
- [x] My PR is targeted at the **dev branch** (and not towards the master branch).
- [x] I ran the [CODE CHECKS](https://github.com/neuropsychology/NeuroKit/blob/master/.github/CONTRIBUTING.rst#run-code-checks) on the files I added or modified and fixed the errors.
- [ ] I have added the newly added features to **News.rst** (if applicable)

[1] J. Emrich, A. Gargano, T. Koka and M. Muma, "Physiology-Informed ECG Delineation Based on Peak Prominence," 2024 32nd European Signal Processing Conference (EUSIPCO), Lyon, France, 2024, pp. 1402-1406, doi: 10.23919/EUSIPCO63174.2024.10715353.

### CODE REVIEW

File: neurokit2/ecg/ecg_delineate.py [hunk 1]
```diff
@@ -49,8 +49,8 @@ def ecg_delineate(
     sampling_rate : int
         The sampling frequency of ``ecg_signal`` (in Hz, i.e., samples/second). Defaults to 1000.
     method : str
-        Can be one of ``"peak"`` for a peak-based method, ``"cwt"`` for continuous wavelet transform
-        or ``"dwt"`` (default) for discrete wavelet transform.
+        Can be one of ``"peak"`` for a peak-based method, ``"prominence"`` for a peak-prominence-based method,
```
Comment (DominiqueMakowski) [inline neurokit2/ecg/ecg_delineate.py:52]:
```suggestion
        Can be one of ``"peak"`` for a peak-based method, ``"prominence"`` for a peak-prominence-based method,
```
```suggestion
        Can be one of ``"peak"`` for a peak-based method, ``"prominence"`` for a peak-prominence-based method (Emrich et al., 2024),
```

File: neurokit2/ecg/ecg_delineate.py [hunk 2]
```diff
@@ -739,6 +752,195 @@ def _ecg_delineator_cwt(ecg, rpeaks=None, sampling_rate=1000):
     }
 
 
+# =============================================================================
+# PROMINENCE METHOD
```
Comment (DominiqueMakowski) [inline neurokit2/ecg/ecg_delineate.py:756]:
```suggestion
# PROMINENCE METHOD (Emrich et al., 2024)
```

File: neurokit2/ecg/ecg_delineate.py [hunk 3]
```diff
@@ -49,8 +49,8 @@ def ecg_delineate(
     sampling_rate : int
         The sampling frequency of ``ecg_signal`` (in Hz, i.e., samples/second). Defaults to 1000.
     method : str
-        Can be one of ``"peak"`` for a peak-based method, ``"cwt"`` for continuous wavelet transform
-        or ``"dwt"`` (default) for discrete wavelet transform.
+        Can be one of ``"peak"`` for a peak-based method, ``"prominence"`` for a peak-prominence-based method (Emrich et al., 2024),
```
Comment (DominiqueMakowski) [inline neurokit2/ecg/ecg_delineate.py:52]: I think CI complains about the line length here

```suggestion
        Can be one of ``"peak"`` for a peak-based method, ``"prominence"`` for a peak-prominence-based 
        method (Emrich et al., 2024), ...
```

**Comment by JonasEmrich** [comment]
> Out of curiosity, do you have some data / info about how this compares to the other methods in terms of accuracy and speed?

For processing a 30-minute single-lead ECG on a Ryzen 7 1800X, the median runtime was: DWT at 9.86 ± 1.07 s, Peak at 11.86 ± 1.83 s, and Prominence at 1.36 ± 0.75 s (excluding R-peak detection and measuring only the time of delineation).

Detection performance is always dependent on the data and evaluation criteria. Using the standard evaluation procedure from the literature (measuring the averaged min-errors over all leads), the prominence-based delineator achieved better F1-scores and lower error variability. When combining multiple leads a further performance gain can be expected. 
However, detecting onsets and offsets still remains challenging. On clean and simulated data, other methods may achieve lower detection errors for these features, while the prominence method performed better in terms of F1-score and error-variability when we applied it to real-world datasets including diverse morphologies and greater variability.

**Comment by DominiqueMakowski** [comment]
Interesting, thanks. Please don't hesitate to let me know in the future - once maybe you gather even more evidence in favour of its superiority - if you think it should become the new default (assuming it is indeed an overall improvement for *most* use-cases)

Related, maybe we could add in the docstrings a sentence or two to help the user pick the method. Like e.g., "The prominence method might be useful to detect the waves, but the peak method might be more precise for detecting the onsets/offsets of the waves (but is very sensitive to signal quality and noise)" what do you think?

**Comment by JonasEmrich** [comment]
Great Idea! I tried to make a fitting suggestion

File: neurokit2/ecg/ecg_delineate.py [hunk 4]
```diff
@@ -49,8 +49,12 @@ def ecg_delineate(
...
         The sampling frequency of ``ecg_signal`` (in Hz, i.e., samples/second). Defaults to 1000.
     method : str
-        Can be one of ``"peak"`` for a peak-based method, ``"cwt"`` for continuous wavelet transform
-        or ``"dwt"`` (default) for discrete wavelet transform.
+        Can be one of ``"peak"`` for a peak-based method, ``"prominence"`` for a peak-prominence-based
+        method (Emrich et al., 2024), ``"cwt"`` for continuous wavelet transform or ``"dwt"`` (default)
+        for discrete wavelet transform.
+        The ``"prominence"`` method might be useful to detect the waves, allowing to set individual physiological
+        limits (see kwargs), while the ``"dwt"`` method might be more precise for detecting the onsets and offsets
+        of the waves (but might exhibit lower accuracy when there is significant variation in wave morphology).
```
Comment (DominiqueMakowski) [inline neurokit2/ecg/ecg_delineate.py:57]:
```suggestion
        of the waves (but might exhibit lower accuracy when there is significant variation in wave morphology).
        The ``"peak"`` method, which uses the zero-crossings of the signal derivatives, works best with very clean signals.
```

**Comment by DominiqueMakowski** [comment]
Sweet, thanks again @JonasEmrich, I think we're good to go 🚀

**Pull request merged**